src/main/java/com/direwolf20/buildinggadgets/client/gui/materiallist/ScrollingMaterialList.java

```diff
@@ -1,214 +1,235 @@
+package com.direwolf20.buildinggadgets.client.gui.materiallist;
+
+import com.direwolf20.buildinggadgets.api.building.view.*;
+import com.direwolf20.buildinggadgets.api.exceptions.TemplateException;
+import com.direwolf20.buildinggadgets.api.materials.MaterialList;
+import com.direwolf20.buildinggadgets.api.materials.UniqueItem;
+import com.direwolf20.buildinggadgets.api.template.IBuildOpenOptions;
+import com.direwolf20.buildinggadgets.api.template.SimpleBuildOpenOptions;
+import com.direwolf20.buildinggadgets.client.gui.base.EntryList;
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.util.helpers.InventoryHelper;
+import com.google.common.collect.Multiset;
+import com.mojang.blaze3d.platform.GlStateManager;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.entity.player.ClientPlayerEntity;
+import net.minecraft.client.gui.widget.list.ExtendedList;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+import org.lwjgl.glfw.GLFW;
+
+import java.awt.*;
+import java.util.Objects;
+
+import static com.direwolf20.buildinggadgets.client.gui.materiallist.MaterialListGUI.*;
+import static com.direwolf20.buildinggadgets.client.gui.materiallist.ScrollingMaterialList.Entry;
+import static org.lwjgl.opengl.GL11.*;
+
+class ScrollingMaterialList extends EntryList<Entry> {
+
+    static final int TOP = 16;
+    static final int BOTTOM = 32;
+
+    private static final int SLOT_SIZE = 18;
+    private static final int MARGIN = 2;
+    private static final int ENTRY_HEIGHT = Math.max(SLOT_SIZE + MARGIN * 2, Minecraft.getInstance().fontRenderer.FONT_HEIGHT * 2 + MARGIN * 3);
+    private static final int LINE_SIDE_MARGIN = 8;
+
+    private MaterialListGUI gui;
+
+    private SortingModes sortingMode;
+
+    public ScrollingMaterialList(MaterialListGUI gui) {
+        super(gui.getWindowLeftX(),
+                gui.getWindowTopY() + TOP,
+                gui.getWindowWidth(),
+                gui.getWindowHeight() - TOP - BOTTOM,
+                ENTRY_HEIGHT);
+        this.gui = gui;
+
+        World world = Minecraft.getInstance().world;
+        ClientPlayerEntity player = Minecraft.getInstance().player;
+        IBuildContext context = SimpleBuildContext.builder()
+                .buildingPlayer(player)
+                .usedStack(gui.getTemplateItem())
+                .build(world);
+        IBuildOpenOptions options = SimpleBuildOpenOptions.withContext(context);
+        try (IBuildView view = Objects.requireNonNull(gui.getTemplateCapability().createViewInContext(options))) {
+            MaterialList materialList = gui.getTemplateHeader().getRequiredItems();
+            if (materialList == null) {
+                materialList = view.estimateRequiredItems();
+            }
+            Multiset<UniqueItem> materials = materialList.getRequiredItems();
```

Comment:
Since the most recent changes, where I finished the MaterialList System this is no longer the full set of items...

- `MaterialList#getRequiredItems()` is the required set of Items, which is independent of the chosen option.
- `MaterialList#getItemOptions()` returns an evaluated set of options from which one must be true, for this MaterialList to be satisfied.
- You can iterate over the `MaterialList` to get all options combined with the constant requiredItems in a single ImmutableMultiset.

Could you change it, so that the options are displayed in some way? I presume that it would be the easiest for you to iterate over the MaterialList (for example one iteration step a second, and when reached the end of the iteration, just reset it...), as you don't need to worry about combining the sets etc.